File: DTC.Api/Dtos/Group/CreateGroupDto.cs
using System.ComponentModel.DataAnnotations;

namespace DTC.Api.Dtos.Group
{
    public class CreateGroupDto
    {
        [Range(1, int.MaxValue, ErrorMessage = "Das Turnier ist ein Pflichtfeld.")]
        public int TournamentId { get; set; }

        [Range(1, int.MaxValue, ErrorMessage = "Die Reihenfolge muss größer als 0 sein.")]
        public int Sequence { get; set; }

        [Required(ErrorMessage = "Der Gruppenname ist ein Pflichtfeld.")]
        [StringLength(100, ErrorMessage = "Der Gruppenname darf maximal 100 Zeichen lang sein.")]
        public string Name { get; set; } = string.Empty;

        [Range(1, int.MaxValue, ErrorMessage = "Die Anzahl der Qualifikanten muss größer als 0 sein.")]
        public int QualifiersCount { get; set; }
    }
}

File: dtc.frontend/src/dtos/group/CreateGroupDto.ts
import z from "zod";

export const createGroupDtoSchema = z.object({
  tournamentId: z.number().int().min(1),
  sequence: z.number().int().min(1),
  name: z.string().min(1).max(100),
  qualifiersCount: z.number().int().min(1),
});

export type CreateGroupDto = z.infer<typeof createGroupDtoSchema>;
